ui/src/app/views/account-creation/component.ts

```diff
@@ -145,4 +167,83 @@ export class AccountCreationComponent {
     return (/[A-Z]/.test(str));
   }
 
+  leaveFocusUsername(): void {
+    this.usernameOffFocus = true;
+  }
+
+  enterFocusUsername(): void {
+    this.usernameOffFocus = false;
+  }
+
+  get isUsernameValidationError(): boolean {
+    return this.usernameConflictError || this.usernameInvalidError;
+  }
+
+  get showUsernameValidationError(): boolean {
+    if (isBlank(this.profile.username) || !this.usernameOffFocus) {
+      return false;
+    }
+    return this.isUsernameValidationError;
+  }
+
+  get showUsernameValidationSuccess(): boolean {
+    if (isBlank(this.profile.username) || !this.usernameOffFocus) {
+      return false;
+    }
+    return !this.isUsernameValidationError;
+  }
+
+  leaveFocusPassword(): void {
+    this.passwordOffFocus = true;
+  }
+
+  enterFocusPassword(): void {
+    this.passwordOffFocus = false;
+  }
+
+  get isPasswordValidationError(): boolean {
+    return (this.showPasswordLengthError ||
+    this.containsLowerAndUpperError);
+  }
+
+  get showPasswordValidationError(): boolean {
+    if (isBlank(this.password) || !this.passwordOffFocus) {
+      return false;
+    }
+    return this.isPasswordValidationError;
+  }
+
+  get showPasswordValidationSuccess(): boolean {
+    if (isBlank(this.password) || !this.passwordOffFocus) {
+      return false;
+    }
+    return !this.isPasswordValidationError;
+  }
+
+  leaveFocusPasswordAgain(): void {
+    this.passwordAgainOffFocus = true;
+  }
+
+  enterFocusPasswordAgain(): void {
+    this.passwordAgainOffFocus = false;
+  }
+
+  get isPasswordAgainValidationError(): boolean {
+    return this.showPasswordsDoNotMatchError;
```

Comment:
Seems slightly wrong that this would call into this "show" method - is this right?